editor/components/block-list/block.js

```diff
@@ -168,6 +169,33 @@ class BlockListBlock extends Component {
 		this.props.blockRef( node, this.props.uid );
 	}
 
+	handleFocusOutside( event ) {
+		if ( ! this.props.isSelected ) {
+			return;
+		}
+
+		// There are several cases in which focus transitions outside the DOM
+		// space of a block, but is not truly outside, typically when managing
+		// toolbar or inspector controls. While this implementation binds
+		// awareness to ancestry, it still retains more self-sufficiency to a
+		// block than attempting to externally manage focus transitions. An
+		// alternative here would leverage React 16 portal virtual event
+		// bubbling, but we must first eliminate dependencies between toolbar
+		// and multi-selection DOM-based event bubbling.
+		//
+		// See: https://github.com/WordPress/gutenberg/pull/3083
+		// See: https://reactjs.org/docs/portals.html#event-bubbling-through-portals
+		const { relatedTarget } = event;
+		const isOutside = ! relatedTarget || ! relatedTarget.closest( [
+			'.editor-header',
+			'.editor-sidebar',
```

Comment:
This component is reusable and these two selectors are not guaranteed to exist (example P2). Should we use a prop for these or a setting in the context?

Reply:
>This component is reusable and these two selectors are not guaranteed to exist (example P2). Should we use a prop for these or a setting in the context?

I agree, and noted in Implementation Notes that this was not ideal, but slightly improved from the previous implementation. I tried an alternate approach in 00fcfca5415d0250319749f5c71cc301afe5564c, passing an `onDeselectBlock` callback from the visual editor to control deselection via `event.preventDefault`. What are your thoughts on this approach?